src/main/java/graphql/execution/SimpleExecutionStrategy.java

```diff
@@ -49,6 +51,6 @@ public ExecutionResult execute(ExecutionContext executionContext, ExecutionStrat
                 break;
             }
         }
-        return new ExecutionResultImpl(results, executionContext.getErrors());
+        return CompletableFuture.completedFuture(new ExecutionResultImpl(results, executionContext.getErrors()));
```

Comment:
Notice here how we have a materialised ExecutionResult and then we wrap into a completed Future
so that to the outside it looks like asynch promise?

I am not sure of the value here.  It allows `thenApply` style composition but it nots async all the way down.

That said I guess it allows other strategies to be more or less asynch and hence the signature is in the right shape.

Actually as I type this, it makes me see how this is useful